Src/IdiomsUnitTest/ConstructorInitializedMemberAssertionTest.cs

```diff
@@ -475,6 +474,84 @@ public void VerifyTypeWithPublicReadOnlyFieldsIncorrectlyInitialisedViaConstruct
             // Teardown
         }
 
+        [Fact]
+        public void VerifyWellBehavedEnumFieldDoesNotThrow()
+        {
+            // Fixture setup
+            var dummyComposer = new Fixture();
+            var sut = new ConstructorInitializedMemberAssertion(dummyComposer);
+            var fieldInfo = typeof(ReadOnlyFieldInitializedViaConstructor<TestEnum>).GetField("Field");
+
+            // Exercise system and verify outcome
+            Assert.DoesNotThrow(() => sut.Verify(fieldInfo));
+            // Teardown
+        }
+
+        [Fact]
+        public void VerifyIllBehavedEnumFieldDoesThrow()
+        {
+            // Fixture setup
+            var dummyComposer = new Fixture();
+            var sut = new ConstructorInitializedMemberAssertion(dummyComposer);
+            var fieldInfo = typeof(ReadOnlyFieldIncorrectlyInitializedViaConstructor<TestEnum>).GetField("Field");
+
+            // Exercise system and verify outcome
+            Assert.Throws<ConstructorInitializedMemberException>(() => sut.Verify(fieldInfo));
+            // Teardown
+        }
+
+        [Fact]
+        public void VerifyWellBehavedEnumPropertyDoesThrow()
+        {
+            // Fixture setup
+            var dummyComposer = new Fixture();
+            var sut = new ConstructorInitializedMemberAssertion(dummyComposer);
+            var propertyInfo = typeof(ReadOnlyPropertyInitializedViaConstructor<TestEnum>).GetProperty("Property");
+
+            // Exercise system and verify outcome
+            Assert.DoesNotThrow(() => sut.Verify(propertyInfo));
+            // Teardown
+        }
+
+        [Fact]
+        public void VerifyIllBehavedEnumPropertyDoesThrow()
+        {
+            // Fixture setup
+            var dummyComposer = new Fixture();
+            var sut = new ConstructorInitializedMemberAssertion(dummyComposer);
+            var propertyInfo = typeof(ReadOnlyPropertyIncorrectlyInitializedViaConstructor<TestEnum>).GetProperty("Property");
+
+            // Exercise system and verify outcome
+            Assert.Throws<ConstructorInitializedMemberException>(() => sut.Verify(propertyInfo));
+            // Teardown
+        }
+
+        [Fact]
+        public void VerifyDefaultOnlyEnumFieldDoesThrow()
+        {
+            // Fixture setup
+            var dummyComposer = new Fixture();
+            var sut = new ConstructorInitializedMemberAssertion(dummyComposer);
+            var fieldInfo = typeof(ReadOnlyFieldIncorrectlyInitializedViaConstructor<TestDefaultOnlyEnum>).GetField("Field");
+
+            // Exercise system and verify outcome
+            Assert.Throws<ConstructorInitializedMemberException>(() => sut.Verify(fieldInfo));
+            // Teardown
+        }
+
+        [Fact]
+        public void VerifyDefaultOnlyEnumPropertyDoesThrow()
```

Comment:
Tests were refactored and are now relying on parameterised input. The result is very nice.